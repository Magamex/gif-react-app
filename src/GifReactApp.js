import React, { Fragment,useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifReactApp = ({ defaultCategories = [] }) => {

  // const [categories, setCategories] = useState(['One Punch'])
  const [categories, setCategories] = useState(defaultCategories)

  const handleAdd = () => {
    setCategories([...categories,'Pokemon'])
    // setCategories( cat => [...cat,'Pokemon'])
  }

  return (
    <Fragment>
        <h2>GifReactApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr/>

        {/* <button onClick={handleAdd}>Agregar</button> */}

        <ol>
          {
            categories.map(category =>(
              <GifGrid 
                key={category} 
                category={category}
                />
            ))
          }
        </ol>

    </Fragment>
  )
}
