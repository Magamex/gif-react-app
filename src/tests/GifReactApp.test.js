import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import { GifReactApp } from '../GifReactApp';

describe('Test en GifReactApp', () => {
    test('Debe generarse correctamente', () => {
        const wrapper = shallow(<GifReactApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categorias = ['One Punch','Batman']
        
        const wrapper = shallow(<GifReactApp defaultCategories={categorias}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    });
});