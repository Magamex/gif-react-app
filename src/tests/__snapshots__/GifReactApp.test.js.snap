// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en GifReactApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifReactApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Batman"
      key="Batman"
    />
  </ol>
</Fragment>
`;

exports[`Test en GifReactApp Debe generarse correctamente 1`] = `
<Fragment>
  <h2>
    GifReactApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
